devicetypes/smartthings/zooz-4-in-1-sensor.src/zooz-4-in-1-sensor.groovy

```diff
@@ -236,3 +282,63 @@ private secure(physicalgraph.zwave.Command cmd) {
 private secureSequence(commands, delay = 200) {
 	delayBetween(commands.collect{ secure(it) }, delay)
 }
+
+def getConfigurationCommands() {
+	log.debug "getConfigurationCommands"
+	def result = []
+	if (isEverspringSP815()) {
+		if (!state.temperatureAndHumidityReport) state.temperatureAndHumidityReport = getEverspringDefaults[1]
+		if (!state.retriggerIntervalSettings) state.retriggerIntervalSettings = getEverspringDefaults[2]
+		Integer temperatureAndHumidityReport = (settings.temperatureAndHumidityReport as Integer) ?: everspringDefaults[1]
+		Integer retriggerIntervalSettings = (settings.retriggerIntervalSettings as Integer) ?: everspringDefaults[2]
+		if (!isConfigured() || (temperatureAndHumidityReport != state.temperatureAndHumidityReport || retriggerIntervalSettings != state.retriggerIntervalSettings)) {
+			state.configured = false // this flag needs to be set to false when settings are changed (and the device was initially configured before)
+			if (!state.temperatureConfigured || temperatureAndHumidityReport != state.temperatureAndHumidityReport) {
+				state.temperatureConfigured = false
+				result << zwave.configurationV2.configurationSet(parameterNumber: 1, size: 2, scaledConfigurationValue: temperatureAndHumidityReport)
+				result << zwave.configurationV2.configurationGet(parameterNumber: 1)
+			}
+			if (!state.intervalConfigured || retriggerIntervalSettings != state.retriggerIntervalSettings) {
+				state.intervalConfigured = false
+				result << zwave.configurationV2.configurationSet(parameterNumber: 2, size: 2, scaledConfigurationValue: retriggerIntervalSettings)
+				result << zwave.configurationV2.configurationGet(parameterNumber: 2)
+			}
+		}
+	}
+	return result
+}
+
+def getEverspringDefaults() {
+	[1: 600,
+	 2: 180]
+}
+
+def lateConfigure() {
+	log.debug "lateConfigure"
+	sendHubCommand(getConfigurationCommands(),200)
+}
+
+private isConfigured() {
+	return state.configured == true
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	if (isEverspringSP815()) {
+		if (cmd.parameterNumber == 1) {
+			state.temperatureAndHumidityReport = scaledConfigurationValue
+			state.temperatureConfigured = true
+		} else if (cmd.parameterNumber == 2) {
+			state.retriggerIntervalSettings = scaledConfigurationValue
+			state.intervalConfigured = true
+		}
+		if (state.intervalConfigured == true && state.temperatureConfigured== true) {
```

Comment:
Space before the `==`

Reply:
I don't see a point in comparing a bool value with another bool

Reply:
ok, fixed two problems at once

Reply:
@PKacprowiczS Good catch!